### PR TITLE
Login, Logout, Me, and Auth Middleware

### LINKED ISSUE #10: Auth / login Endpoints + JWT + httpOnly cookie
**Is your feature request related to a problem? Please describe.**
Users need to authenticate and establish a session

*Describe the solution you'd like**
- Endpoint: POST /api/auth/login
- Request: { email: string, password: string }
- Query user by email
- Compare password with bcrypt.compare()
- If match: create JWT with { userId, email } payload
- Set httpOnly cookie: Set-Cookie: token=jwt; HttpOnly; Secure; SameSite=Strict; Max-Age=7d
- Return { success: true, user: { id, email } }
- If no match: 401 Unauthorized

**Describe alternatives you've considered**
Could use session storage, but JWT is stateless (better!)

**Additional context**
Use jsonwebtoken package, store the secret in .env, cookie expiration should be 7 days

### PR DESCRIPTION
**Pull request by mpolisch**
## Summary
Implements login, logout, and me endpoints, along with auth middleware to protect authenticated routes. Full JWT session lifecycle defined: login sets httpOnly cookie, middleware validates it on protected routes, logout clears it.

## Approach & Design Decisions
- JWT stored in httpOnly cookie (invisible to JS to prevent XSS) 
- Secure flag toggled by NODE_ENV so localhost works without https
- When a user doesn't exist, bcrypt.compare runs against a dummy hash to keep response time consistent (timing attack prevention)
- COOKIE_NAME & COOKIE_MAX_AGE as constants to ensure the cookie cleared on logout is identical to one used for login
- clearCookie uses sameoptions as setCookie, it must match exactly or the browser won't clear the cookie
- JWT_SECRET is fail-fast
- express.json({ limit: '10kb'}): request body size limited to stop DoS / DDoS attacks
- jest.setup.ts: injects JWT_SECRET before modules are evaluated
- express.d.ts: adds req.user to Express Request interface

## Security Considerations
- JWT signed with JWT_SECRET from env
- httpOnly cookie prevents client-side JS from accessing the token
- SameSite: Strict to mitigate CSRF attacks
- Secure: True in prodction to ensure cookies only sent over HTTPS
- Timing-sage login: bcrypt.compare always runs whether the user exists
- Wrong PW and non-existent email both return 401
- TokenExpiredError is handled separate from other JWT errors, giving users a clear 'session expired' message
- Request body capped at 10kb

## Related Issues
Closes #10 #11

## Testing
**What's covered:**
- [x] Unit tests
- [x] Edge cases (list below)

**Edge cases considered:**
- Non-existent email: bcrypt still runs and throws 401
- Wrong password: same 401 message as non-existent email
- Expired token: TokenExpiredError, returns specific 401 message
- Tampered token: caught by jwt.verify, returns 401
- Missing cookie: caught before jwt.verify, returns 401
- /me after logout returns 401

## Checklist
- [x] No secrets or credentials in code
- [x] Input validation in place
- [x] Error handling covers failure paths
- [x] Code is self-documenting or commented where non-obvious
- [x] No dead code or debug logs left in

## Notes
- req.user! non-null assertion in /me controller is safe since the route is always behind requireAuth middleware, ensuring req.user is set before the handler runs
- No integration tests (out of scope for this project) but all behaviours were tested via Postman

### CODE REVIEW

File: src/services/auth.ts [hunk 1]
```diff
@@ -23,3 +27,36 @@ export const registerUser = async (data: RegisterDTO): Promise<PublicUser | null
...
+        'SELECT id, email, password_hash, created_at, updated_at FROM users WHERE email = $1',
+        [email],
+    );
+
+    const user = result.rows[0];
+
+    // Use a constant-time compare even when user doesn't exist to prevent
+    // timing attacks that could reveal whether an email is registered.
+    const passwordHash = user?.password_hash ?? '$2b$12$invalidhashfortimingattackprevention';
+    const isValid = await bcrypt.compare(password, passwordHash);
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply src/services/auth.ts:44]: I tested it with an invalid email and it still threw a 401 with ~230ms response time, tested with valid email & invalid password, and it also threw 401 with ~230ms response time.

File: src/middleware/auth.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import type { Request, Response, NextFunction } from 'express';
+import jwt from 'jsonwebtoken';
+import { AppError } from '../utils/AppError.js';
+import type { AuthPayload } from '../types/index.js';
+
+const JWT_SECRET = process.env.JWT_SECRET as string;
+
+export const requireAuth = (req: Request, res: Response, next: NextFunction): void => {
+    const token = req.cookies?.token as string | undefined;
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply src/middleware/auth.ts]: Resolved.
